rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1824,7 +1830,81 @@ func Create_TerminationInstruction: <"Creates the relevant primitive instruction
+		deliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+		undeliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"This function applies to contractual products only.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+	set instruction -> split -> breakdown: <"Sets the first part of the split to be a termination instruction for the existing trade.">
+		[ Create_PrimitiveInstruction(
+			empty, empty, empty, empty,
+						Create_QuantityChangeInstruction(
+						deliveredPriceQuantity, 
+						QuantityChangeDirectionEnum -> Replace, 
+						tradeState->trade->tradableProduct->tradeLot->lotIdentifier
+			),
+			empty, empty, empty, empty,	empty, empty, empty)]
+	
+	add instruction -> split -> breakdown: <"Sets the second part of the split to be a new contract with a new asset payout.">
+		[
+		Create_PrimitiveInstruction(
+			empty, empty, empty, empty,
+			Create_QuantityChangeInstruction(
+				undeliveredPriceQuantity,
+				QuantityChangeDirectionEnum -> Replace,
+				empty
+				),
+			empty, empty,
+			Create_PartialDeliveryInstructionTermsChange(
+				tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+				effectiveDate, terminationDate
+				),
+			empty, empty, empty, empty
+			)
+		]
+	
+func Create_PartialDeliveryInstructionTermsChange:
+	inputs:
+		contractualProduct ContractualProduct (1..1) <"The contractual product to be modifier.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		terminationDate AdjustableOrRelativeDate (1..1) <"The new termination date.">
+	output:
+		termsChangeInstruction TermsChangeInstruction (1..1)
+
+	set termsChangeInstruction -> product -> contractualProduct:
+		contractualProduct
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> effectiveDate:
+		effectiveDate
+	set termsChangeInstruction -> product -> contractualProduct -> economicTerms -> terminationDate:
+		terminationDate
+
+func Qualify_PartialDelivery: <"Qualification of a partial delivery which constitutes a contract close and reopen with a difference quantity and termination date.">
+	[qualification BusinessEvent]
+	inputs:
+		businessEvent BusinessEvent (1..1)
+	output: is_event boolean (1..1)
+
+	alias beforeEconomicterms: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias openEconomicTerms: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias closedTradeState: FilterClosedTradeStates( businessEvent -> after )
+
+	set is_event:
+		beforeEconomicterms exists and
+		openEconomicTerms exists and
+		closedTradeState count = 1 and
```

Comment:
will the original trade be closed at this point? Will the first instruction in the split close it?

Reply:
Yes